ext/mysql2/client.c

```diff
@@ -880,7 +880,7 @@ static VALUE rb_mysql_client_server_info(VALUE self) {
  *
  * Return the file descriptor number for this client.
  */
-static VALUE rb_mysql_client_socket(VALUE self) {
+static VALUE rb_mysql_client_socket(RB_MYSQL_UNUSED VALUE self) {
```

Comment:
The above `UNUSED(self);` would work and to define two different versions of this function would work equally. I'll update the pull request based on your desire.

As a side-note: In pg.gem I solved the socket issue on Windows years ago: https://github.com/larskanis/ruby-pg/commit/84a1bb7fa40b59001